### ISSUE TITLE
Dom.nodeFilter type is wrong

### ISSUE THREAD

**Issue by Coobaha** [reasonml-community/bs-webapi-incubator#96]
https://bucklescript.github.io/bucklescript/api/Dom.html#TYPEnodeFilter

It is typed as record. 

https://github.com/reasonml-community/bs-webapi-incubator/blob/df1585b0db2fc68d92a0ec5042a075a8d63ba602/tests/dom/nodes/document_test.re#L35

So it compiles to `[() => 0]` instead of `{ acceptNode: () => 0 }`;

**Comment by bobzhang** [comment]
is it correct to type it as `< .. > Js.t` ?

**Comment by Coobaha** [comment]
@bobzhang Yes, please check example from [mdn](https://developer.mozilla.org/en-US/docs/Web/API/NodeIterator/filter)

**Comment by yawaramin** [comment]
This should be fixed with BuckleScript's records-as-objects representation. if not we can reopen.
